opensrp-anc/src/main/java/org/smartregister/anc/library/listener/ContactTaskDisplayClickListener.java

```diff
@@ -0,0 +1,265 @@
+     * @param context   {@link Context}
+     * @return fields  {@link JSONArray}
+     */
+    private JSONArray loadSubFormFields(JSONObject taskValue, Context context) {
+        JSONArray fields = new JSONArray();
+        try {
+            if (taskValue != null && taskValue.has(JsonFormConstants.CONTENT_FORM)) {
+                String subFormName = taskValue.getString(JsonFormConstants.CONTENT_FORM);
+                JSONObject subForm = ContactJsonFormUtils.getSubFormJson(subFormName, "", context);
+                if (subForm.has(JsonFormConstants.CONTENT_FORM)) {
+                    fields = subForm.getJSONArray(JsonFormConstants.CONTENT_FORM);
+                }
+            }
+        } catch (JSONException e) {
+            Timber.e(e, " --> loadSubFormFields");
+        } catch (Exception e) {
+            Timber.e(e, " --> loadSubFormFields");
+        }
+        return fields;
+    }
+
+    /**
+     * Get the form title for the accordion text
+     *
+     * @param taskValue {@link JSONObject}
+     * @return title {@link String}
+     */
+    private String getFormTitle(JSONObject taskValue) {
+        String title = "";
+        if (taskValue != null && taskValue.has(JsonFormConstants.TEXT)) {
+            title = taskValue.optString(JsonFormConstants.TEXT, "");
+        }
+        return title;
+    }
+
+    /**
+     * Loads the main contact tasks form. Returns a JSONObject of the form.
+     *
+     * @param context {@link Context}
+     * @return jsonForm {@link JSONObject}
+     */
+    private JSONObject loadTasksForm(Context context) {
+        JSONObject form = new JSONObject();
+        try {
+            FormUtils formUtils = new FormUtils(context);
+            form = formUtils.getFormJson(ConstantsUtils.JsonFormUtils.ANC_TEST_TASKS);
+        } catch (Exception e) {
+            Timber.e(e, " --> loadTasksForm");
+        }
+        return form;
+    }
+
+    /**
+     * Updates the form step1 title to match the test header
+     *
+     * @param form  {@link JSONObject}
+     * @param title {@link String}
+     */
+    private void updateFormTitle(JSONObject form, String title) {
+        try {
+            if (form != null && StringUtils.isNotBlank(title)) {
+                if (form.has(JsonFormConstants.STEP1)) {
+                    JSONObject stepOne = form.getJSONObject(JsonFormConstants.STEP1);
+                    stepOne.put(JsonFormConstants.STEP_TITLE, title);
+                }
+            }
+        } catch (JSONException e) {
+            Timber.e(e, " --> updateFormTitle");
+        }
+    }
+
+    /**
+     * Add the sub form fields to the main form for loading.
+     *
+     * @param form   {@link JSONObject}
+     * @param fields {@link JSONArray}
+     */
+    private void updateFormFields(JSONObject form, JSONArray fields) {
+        try {
+            if (form != null && fields != null && fields.length() > 0) {
+                if (form.has(JsonFormConstants.STEP1)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=4809639)